src/Kdyby/Doctrine/DI/OrmExtension.php

```diff
@@ -417,20 +417,20 @@ protected function processEntityManager($name, array $defaults)
 			$entityManager->addSetup('?->bindEntityManager(?)', [$this->prefix('@' . $name . '.diagnosticsPanel'), '@self']);
 		}
 
-		if ($isDefault && $config['defaultRepositoryClassName'] === 'Kdyby\Doctrine\EntityDao') {
+		if ($isDefault && $config['defaultRepositoryClassName'] === 'Kdyby\Doctrine\EntityRepository') {
 			// syntax sugar for config
-			$builder->addDefinition($this->prefix('dao'))
-				->setClass('Kdyby\Doctrine\EntityDao')
-				->setFactory('@Kdyby\Doctrine\EntityManager::getDao', [new Code\PhpLiteral('$entityName')])
+			$builder->addDefinition($this->prefix('repository'))
+				->setClass($config['defaultRepositoryClassName'])
+				->setFactory('@Kdyby\Doctrine\EntityManager::getRepository', [new Code\PhpLiteral('$entityName')])
 				->setParameters(['entityName'])
 				->setInject(FALSE);
 
 			// interface for models & presenters
-			$builder->addDefinition($this->prefix('daoFactory'))
-				->setClass('Kdyby\Doctrine\EntityDao')
-				->setFactory('@Kdyby\Doctrine\EntityManager::getDao', [new Code\PhpLiteral('$entityName')])
-				->setParameters(['entityName'])
-				->setImplement('Kdyby\Doctrine\EntityDaoFactory')
+			$builder->addDefinition($this->prefix('repositoryFactory'))
+				->setClass($config['defaultRepositoryClassName'])
+				->setImplement('Kdyby\Doctrine\DI\IRepositoryFactory')
+				->setArguments([new Code\PhpLiteral('$entityManager'), new Code\PhpLiteral('$classMetadata')])
+				->setParameters(['Doctrine\ORM\EntityManagerInterface entityManager', 'Doctrine\ORM\Mapping\ClassMetadata classMetadata'])
 				->setInject(FALSE)->setAutowired(TRUE);
 		}
 
```

Comment:
under this line, there is `...defaultRepositoryFactory` service definition, can you please change `->setAutowired(FALSE)` to `->setAutowired($isDefault)` ?